app/models/product_import/entry_processor.rb

```diff
@@ -185,6 +185,7 @@ def save_new_product(entry)
     def save_variant(entry)
       variant = entry.product_object
       variant.import_date = @import_time
+      variant.assign_units
```

Comment:
I see that [this happens `before_save`](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/models/spree/variant.rb#L84-L87). Was the issue that it needs to happen before validation? If so, perhaps instead we need to update the model to run it `before_validation`.

Reply:
I was curious about it as well, so I gave it a go. 
Moving `assign_units` to a `before_validation` callback fixes the the problem with the import. But according to the specs,  the `before_save` callback is there for a reason and the variant specs fails without it.
Running both a `before_validation` and `before_save callback` doesn't work either ( and it feels wrong to do that).

Reply:
Thanks heaps! That resolves the last of my queries.